### PR TITLE
[SPARK-12199][DOC] Follow-up: Refine example code in ml-features.md

### PR DESCRIPTION
**Pull request by yinxusen**
https://issues.apache.org/jira/browse/SPARK-12199

Follow-up PR of SPARK-11551. Fix some errors in ml-features.md

@mengxr

### CODE REVIEW

**Comment by mengxr** [comment]
@yinxusen Could you fix conflicts? Thanks!

**Comment by yinxusen** [comment]
@mengxr The conflict is caused by the revert of https://github.com/apache/spark/pull/10002, for unchecking of Python style. I think we need to wait for that PR being merged. I have sent a PR to @somideshmukh and hope he could merge it ASAP.

**Comment by mengxr** [comment]
test it please

**Comment by yinxusen** [comment]
retest it please

**Comment by yinxusen** [comment]
@mengxr Does it look good to you? :)

**Comment by BenFradet** [comment]
There is one nit in [polynomial_expansion_example.py](https://github.com/apache/spark/blob/master/examples/src/main/python/ml/polynomial_expansion_example.py#L33) with the spaces.
Except that, it lgtm.

**Comment by BenFradet** [comment]
Plus, there is inconsistent naming between the docs and the example file for Element[W|w]iseProductExample.scala

**Comment by yinxusen** [comment]
@BenFradet All clear now. Thanks!

**Comment by yinxusen** [comment]
Faint! OSX's file name is not case sensitive... that's why I miss the Element[W|w]iseProductExample.scala error.

**Comment by BenFradet** [comment]
No problem, I'm just glad it's fixed.

**Comment by BenFradet** [comment]
Pinging @mengxr and @jkbradley as it is not curre ntly possible to generate this doc on master.

File: docs/ml-features.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ the [IDF Python docs](api/python/pyspark.ml.html#pyspark.ml.feature.IDF) for mor
 `Word2VecModel`. The model maps each word to a unique fixed-size vector. The `Word2VecModel`
 transforms each document into a vector using the average of all words in the document; this vector
 can then be used for as features for prediction, document similarity calculations, etc.
-Please refer to the [MLlib user guide on Word2Vec](mllib-feature-extraction.html#Word2Vec) for more
+Please refer to the [MLlib user guide on Word2Vec](mllib-feature-extraction.html#word2Vec) for more
```
Comment (jkbradley) [inline docs/ml-features.md]: The "V" in "word2vec" in the link should be lowercase

**Comment by jkbradley** [comment]
I'll make sure this gets in.  Just had 1 comment (and trying to generate the docs locally now)

**Comment by jkbradley** [comment]
OK, it worked locally.  This should be fine after that final comment.

**Comment by jkbradley** [comment]
Since this is a problem for the release, I'll go ahead and merge this with master and branch-1.6.  @yinxusen Could you please send a follow-up to fix the broken link?  Thanks!

**Pull request closed without merging**